httpbin/handlers_test.go

```diff
@@ -1163,16 +1163,32 @@ func TestResponseHeaders(t *testing.T) {
 
 		req, _ := http.NewRequest("GET", fmt.Sprintf("%s/response-headers?%s", srv.URL, wantHeaders.Encode()), nil)
 		resp := must.DoReq(t, client, req)
-		result := mustParseResponse[http.Header](t, resp)
+
+		type bodyResponse struct {
+			Foo           string   `json:"Foo"`
+			Bar           []string `json:"Bar"`
+			ContentType   string   `json:"Content-Type"`
+			ContentLength int      `json:"Content-Length"`
+		}
+
+		result := mustParseResponse[bodyResponse](t, resp)
+
+		assert.ContentType(t, resp, result.ContentType)
+		assert.Equal[bool](t, result.ContentLength > 0, true, "JSON response Content-Length mismatch")
```

Comment:
I think we can do away with most of this specialization if we simplify the response as requested above.